pandas/_libs/tslibs/parsing.pyx

```diff
@@ -1016,9 +1016,11 @@ def guess_datetime_format(dt_str: str, bint dayfirst=False) -> str | None:
 
 cdef str _fill_token(token: str, padding: int):
     cdef str token_filled
-    if "." not in token:
+    if re.search(r"\d*\.\d+", token) is None:
```

Comment:
is the leading `\d*` needed here?

Reply:
`\d+` would be better actually, thanks

This is for when it corresponds with the `%S.%f` directive